packages/pigeon/lib/objc_generator.dart

```diff
@@ -79,6 +79,7 @@ const Map<String, String> _objcTypeForDartTypeMap = <String, String>{
   'Float64List': 'FlutterStandardTypedData',
   'List': 'NSArray',
   'Map': 'NSDictionary',
+  'Object': 'NSObject',
```

Comment:
The ergonomics of this will be poor for developers, because it will require casting to use the values. I know it'll complicate the generator due to the lack of pointer, but we should really translate `Object` to `id`, not `NSObject*`.

E.g., we want people using `Map<String, Object>` to get `NSDictionary<NSString*, id>*` so that they can write things like:
```objc
  BOOL foo = [aMap[@"someBoolKey"] boolValue];
```

rather than having to do:
```objc
  BOOL foo = [((NSSumber*)aMap[@"someBoolKey"]) boolValue];
```

Reply:
Done.